qek/data/extractors.py

```diff
@@ -645,8 +663,14 @@ def _ingest(self, batches: dict[str, Batch]) -> None:
             assert len(batch.jobs) == 1
             for _, job in batch.jobs.items():
                 if job.status == "DONE":
-                    state_dict = job.result
-                    assert state_dict is not None
+                    state_dict = self._state_extractor(job, compiled.sequence)
+                    if state_dict is None:
+                        logger.warning(
+                            "Batch %s (graph %s) did not return a usable state, skipping",
```

Comment:
What's the rationale again for not using string formatting here ?

Reply:
Generally a good practice in logging libraries: don't build strings before knowing whether they'll actually be displayed.